zerver/lib/actions.py

```diff
@@ -2918,8 +2917,9 @@ def internal_prep_private_message(
 
 
 def internal_send_private_message(
-    realm: Realm, sender: UserProfile, recipient_user: UserProfile, content: str
+    sender: UserProfile, recipient_user: UserProfile, content: str
 ) -> Optional[int]:
+    realm = recipient_user.realm
```

Comment:
Can you run the backend test suite with an `assert realm == recipient_user.realm` here and make sure that passes?  

I don't think this statement is theoretically always true given the existence of system bots -- e.g. I think `internal_send_private_message` where the recipient is WELCOME_BOT would not work, and I'm not 100% confident one can't reach that code path (E.g. maybe via outgoing webhook bots or the like).

But it seems likely to be true once we migrate system bots, which I imagine is the project this is a prep for?

Reply:
There were two incorrectly written tests that were causing the assertion to break, but real production code doesn't seem to cause, at least as far as test coverage goes. I fixed the two tests in a preliminary commit. Also added a commit that tests that internal_send_private_message still work for sending to  cross-realm bots even after the argument removal, so we should be good I think?

Correct, this is some cleanup I'm doing on the way to migrating the bots, but might be mergeable independently I think